src/com/tictacgo/data/Board.java

```diff
@@ -283,144 +306,122 @@ public void nextTurn() {
-	 * @return An ArrayList of the collisions.
-	 * 	For each collision:
-	 * 		If its length is 2, we should switch the isX values.
-	 * 		If its length is greater than 2, we should remove the Pieces.
 	 */
-	public ArrayList<ArrayList<Piece>> getHalfwayCollisions(ArrayList<Piece> temp) {
-		ArrayList<ArrayList<Piece>> collisions = new ArrayList<>();
-		for (int i = 0; i < temp.size() - 1; i++) { //For each Piece
-			collisions.add(new ArrayList<Piece>()); //Will be the Pieces that collide with temp(i)
-			collisions.get(collisions.size() - 1).add(temp.get(i)); //Add Piece temp(i)
-			for (int j = i + 1; j < temp.size(); j++) { //For every Piece after i in temp.
-														//This eliminates having two ArrayLists for one collisions
+	private void resolveHalfwayCollisions() {
+		for (int i = 0; i < pieces.size() - 1; i++) {
+      List<Piece> collision = new ArrayList<>(4);
+			collision.add(pieces.get(i));
+			for (int j = i + 1; j < pieces.size(); j++) { //For every Piece after the current one.
 				/**
 				 * X values are the same.
 				 * Y values cross.
 				 */
-				if ((temp.get(i).getLastXPosition() == temp.get(j).getLastXPosition() &&
-					 temp.get(i).getXPosition() == temp.get(j).getXPosition() &&
-					 temp.get(i).getLastYPosition() == temp.get(j).getYPosition() &&
-					 temp.get(i).getYPosition() == temp.get(j).getLastYPosition()) ||
+				if ((pieces.get(i).getLastXPosition() == pieces.get(j).getLastXPosition() &&
+					 pieces.get(i).getXPosition() == pieces.get(j).getXPosition() &&
+					 pieces.get(i).getLastYPosition() == pieces.get(j).getYPosition() &&
+					 pieces.get(i).getYPosition() == pieces.get(j).getLastYPosition()) ||
 				
 				/**
 				 * X values cross.
 				 * Y values are the same.
 				 */
-					(temp.get(i).getLastXPosition() == temp.get(j).getXPosition() &&
-					 temp.get(i).getXPosition() == temp.get(j).getLastXPosition() &&
-					 temp.get(i).getLastYPosition() == temp.get(j).getLastYPosition() &&
-					 temp.get(i).getYPosition() == temp.get(j).getYPosition()) ||
+					(pieces.get(i).getLastXPosition() == pieces.get(j).getXPosition() &&
+					 pieces.get(i).getXPosition() == pieces.get(j).getLastXPosition() &&
+					 pieces.get(i).getLastYPosition() == pieces.get(j).getLastYPosition() &&
+					 pieces.get(i).getYPosition() == pieces.get(j).getYPosition()) ||
 
 				/**
 				 * X values cross.
 				 * Y values cross.
 				 */
-					(temp.get(i).getLastXPosition() == temp.get(j).getXPosition() &&
-					 temp.get(i).getXPosition() == temp.get(j).getLastXPosition() &&
-					 temp.get(i).getLastYPosition() == temp.get(j).getYPosition() &&
-					 temp.get(i).getYPosition() == temp.get(j).getLastYPosition())) {
+					(pieces.get(i).getLastXPosition() == pieces.get(j).getXPosition() &&
+					 pieces.get(i).getXPosition() == pieces.get(j).getLastXPosition() &&
+					 pieces.get(i).getLastYPosition() == pieces.get(j).getYPosition() &&
+					 pieces.get(i).getYPosition() == pieces.get(j).getLastYPosition())) {
```

Comment:
There is definitely a cleaner way to write this, but because it isn't the focus of this PR, I am ignoring it.